app/src/behaviors/behavior_mod_morph.c

```diff
@@ -45,6 +46,8 @@ static int on_mod_morph_binding_pressed(struct zmk_behavior_binding *binding,
     }
 
     if (zmk_hid_get_explicit_mods() & cfg->mods) {
+        zmk_mod_flags_t trigger_mods = zmk_hid_get_explicit_mods() & cfg->mods;
```

Comment:
I patched this, and get a warning that reads that `trigger_mods` is an unused variable. Is this supposed to be used in some sort of restore behavior later?

Also, when using the behavior, if I add shifts to my masked_mods, and my binding has a shift, it still strips them out. I think the masked_mods should mask the trigger_mods and not the resulting binding, i.e. if the result is specified to be `LS(N2)`, it should send `@`, and not the `2` just because `MOD_LSFT` was in the `masked_mods`.

Reply:
> > hey guys, any update on this?
> 
> No real updates. The task list right now is:
> 
>     * [ ]  update patch with latest main
> 
>     * [ ]  fix broken tests
> 
>     * [ ]  add new tests to cover new functionality
> 
> 
> I have not been able to make any progress, cause life priorities.

A quick update on this. I cherry-picked 241f82e into the latest main and successfully build against it. So far, all my tests did go well and `masked_mods` seem to do exactly what they set out to do. In case someone else wants to test without messing with a (minor) merge conflict, this is the commit: e40b9a7.

Reply:
I do not recall such behavior, but I am using an old version of ZMK at the moment.

@urob have you experienced any similar behavior?

Reply:
> I patched this, and get a warning that reads that `trigger_mods` is an unused variable. Is this supposed to be used in some sort of restore behavior later?

This is weird, I just merged this into the latest ZMK and I had no issues.

> Also, when using the behavior, if I add shifts to my masked_mods, and my binding has a shift, it still strips them out. I think the masked_mods should mask the trigger_mods and not the resulting binding, i.e. if the result is specified to be `LS(N2)`, it should send `@`, and not the `2` just because `MOD_LSFT` was in the `masked_mods`.

Unless I am misunderstanding, this is the intended behavior. Don't include `MOD_LSFT` in `masked_mods` if you don't want it to be masked.

Reply:
The problem with the modifier not showing up on the host is that, it gets removed from what I want to send, masked mods should work in the manner of "remove all of these mods from what the user pressed" and everything that's specified inside the bindings should get to the host along with the modifiers that weren't masked.

The idea behind `masked_mods` as it is right now, contradicts the intent of adding a binding that contains any of those modifiers. And the way I understand it, when a user adds stuff to bindings it's because they want that to arrive at the host when pressing the binding position modulated by mod-morph.

Currently the behavior is similar to:
- Calculate which binding we want.
- Add active modifiers to the results
- Add binding selected to the results
- Strip anything in masked_mods from the results
- Send the result to the host.

I think a better approach that would be:
- Calculate which of the bindings we want
- Add active modifiers to the results
- Strip anything in masked_mods from the results
- Add binding selected to the results
- Send the result to the host.

In my proposed flow, there's no "unexpected" deletion of the modifiers explicitly defined in the bindings.

Reply:
> Also, when using the behavior, if I add shifts to my masked_mods, and my binding has a shift, it still strips them out. I think the masked_mods should mask the trigger_mods and not the resulting binding, i.e. if the result is specified to be `LS(N2)`, it should send `@`, and not the `2` just because `MOD_LSFT` was in the `masked_mods`.

For this particular mod-morph, as a workaround, would it work to remove `MOD_LSFT` from the masked mods?